src/utilities/forms.py

```diff
@@ -102,3 +104,42 @@ class Meta:
             # https://code.djangoproject.com/ticket/24453
             'sites': forms.MultipleHiddenInput(),
         }
+
+
+class MenuItemForm(forms.ModelForm):
+    url = forms.RegexField(
+        label=_("URL"),
+        max_length=100,
+        regex=r'^[-\w/\.~]+$',
+        help_text=_("Relative URLs will work too.  E.g. '/courses/ranks/'"),
+        error_messages={
+            "invalid": _(
+                "This value must contain only letters, numbers, dots, "
+                "underscores, dashes, slashes or tildes."
+            ),
+        },
+    )
+
+    class Meta:
+        model = MenuItem
+        fields = '__all__'
+
+    def _trailing_slash_required(self):
+        return (
+            settings.APPEND_SLASH and
+            'django.middleware.common.CommonMiddleware' in settings.MIDDLEWARE
+        )
+
+    def clean_url(self):
```

Comment:
What happens without this?  Aren't these kinds of errors given by default?  Needs docstring explaining why this is needed and what it's doing

Reply:
the functions are necessary to display the correct error messages when a relative url with faulty formatting is submitted
![image](https://user-images.githubusercontent.com/105619909/175722512-35bfdc7f-6ff3-43c0-9268-825fe3742e17.png)
tests and documentation added.